homeassistant/components/iotawatt/sensor.py

```diff
@@ -0,0 +1,127 @@
+            name=coordinator.data["sensors"][ent].getName(),
+        )
+        entities.append(entity)
+
+    async_add_entities(entities)
+
+    async def async_new_entities(sensor_info):
+        """Remove an entity."""
+        ent = sensor_info["entity"]
+        hub_mac_address = sensor_info["mac_address"]
+        name = sensor_info["name"]
+
+        entity = IotaWattSensor(
+            coordinator=coordinator,
+            key=ent,
+            mac_address=hub_mac_address,
+            name=name,
+        )
+        entities = [entity]
+        async_add_entities(entities)
+
+    async_dispatcher_connect(hass, SIGNAL_ADD_DEVICE, async_new_entities)
+
+
+class IotaWattSensor(IotaWattEntity, SensorEntity):
+    """Defines a IoTaWatt Energy Sensor."""
+
+    def __init__(self, coordinator, key, mac_address, name):
+        """Initialize the sensor."""
+        super().__init__(
+            coordinator=coordinator, entity=key, mac_address=mac_address, name=name
+        )
+
+        sensor = self.coordinator.data["sensors"][key]
+        self._ent = key
+        self._name = name
+        self._io_type = sensor.getType()
+        self._attr_state_class = STATE_CLASS_MEASUREMENT
+        self._attr_force_update = True
+
+        unit = sensor.getUnit()
+        if unit == "Watts":
+            self._attr_native_unit_of_measurement = POWER_WATT
+            self._attr_device_class = DEVICE_CLASS_POWER
+        elif unit == "WattHours":
+            self._attr_native_unit_of_measurement = ENERGY_WATT_HOUR
+            self._attr_device_class = DEVICE_CLASS_ENERGY
+            self._attr_state_class = STATE_CLASS_TOTAL_INCREASING
+        elif unit == "Volts":
+            self._attr_native_unit_of_measurement = ELECTRIC_POTENTIAL_VOLT
+            self._attr_device_class = DEVICE_CLASS_VOLTAGE
+        else:
+            self._attr_native_unit_of_measurement = unit
+
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        """Handle updated data from the coordinator."""
+        if self._ent not in self.coordinator.data["sensors"]:
+            entity_registry.async_get(self.hass).async_remove(self.entity_id)
+            return
+
+        super()._handle_coordinator_update()
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes of the device."""
+        if self._io_type == "Input":
+            channel = self.coordinator.data["sensors"][self._ent].getChannel()
+        else:
+            channel = "N/A"
+
+        attrs = {"type": self._io_type, "channel": channel}
+
+        return attrs
+
+    @property
+    def native_value(self):
+        """Return the state of the sensor."""
+        return self.coordinator.data["sensors"][self._ent].getValue()
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return f'IoTaWatt {self._io_type} {self.coordinator.data["sensors"][self._ent].getName()}'
+
+    @property
+    def unique_id(self) -> str:
```

Comment:
You are now mixing properties and `_attr_*`  can this be moved to `self._attr_unique_id` in the init method?

Reply:
I did the same for the ```self._attr_unique_id``` as I did in the above comment. Removed the ```unique_id()``` and moved the property to the ```__init__()``` method. Is that okay?

Reply:
Made the suggested change.